### PR TITLE
Ligo RPC server and related fixes

### PR DESCRIPTION
**Pull request by d4hines**
This PR includes numerous fixes related to encodings, the js libraries and example projects, and the new ligo-deku-rpc server.

### CODE REVIEW

File: deku-c/client/src/deku-c/ligoRpc.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    storage: initialStorage,
+  };
+  const options = {
+    method: "POST",
+    type: "appliction/json",
+    body: JSON.stringify(params),
+  };
+  const compilationQueryString =
+    target === "michelson" ? "?target=michelson" : "";
+  const url = ligoRpc + "/api/v1/compile-contract" + compilationQueryString;
```
Comment (aguillon) [inline deku-c/client/src/deku-c/ligoRpc.ts:49]: something something path.join

Comment (aguillon) [reply deku-c/client/src/deku-c/ligoRpc.ts:49]: or the other one

File: deku-c/client/src/deku-c/index.ts [hunk 2]
```diff
@@ -17,7 +13,6 @@ export type Settings = {
 export class DekuCClient extends DekuPClient {
   readonly ligoRpc?: string;
```
Comment (aguillon) [inline deku-c/client/src/deku-c/index.ts:14]: maybe change name?

File: deku-c/client/src/deku-c/ligoRpc.ts (hunk 1 above)
Comment (d4hines) [reply deku-c/client/src/deku-c/ligoRpc.ts]: Fixed.

File: deku-c/deku-cli/commands/default-parameters.ts [hunk 3]
```diff
@@ -1,2 +1,2 @@
 export const api = "https://deku-canonical-vm0.deku-v1.marigold.dev/";
-export const ligoApi = "http://ligo-deku-rpc.gcp-npr.marigold.dev/";
+export const ligoApi = "http://0.0.0.0:9090"; // FIXME: don't merge with this
```
[the remark this answers is not in the document]
Comment (d4hines) [reply deku-c/deku-cli/commands/default-parameters.ts]: Done.

File: deku-c/client/src/deku-c/index.ts (hunk 2 above)
Comment (d4hines) [reply deku-c/client/src/deku-c/index.ts]: /shrug. Can't think of a better one. Gonna merge with this for now.

File: deku-c/client/src/deku-c/utils.ts [hunk 4]
```diff
@@ -1,101 +1,106 @@
...
   dekuRpc: string,
   hash: string
 ): Promise<string> => {
   const body = { hash };
   const response = await fetch(
-    dekuRpc + "/api/v1/helpers/compute-contract-hash",
+    urlJoin(dekuRpc, "/api/v1/helpers/compute-contract-hash"),
     { method: "POST", body: JSON.stringify(body) }
   );
   const json = await response.json();
```
Comment (Pilou97) [inline deku-c/client/src/deku-c/utils.ts:106]: Shouldn't we check if the status of the fetch response?

**Pull request merged**